### PR TITLE
Remove `AnyThreadEventLoopPolicy`

### LINKED ISSUE #5309: `get_client` within a task creates a new (unused) event loop for every thread
`get_client` tries to check if the current event loop is the same as the worker's event loop, calling tornado `IOLoop.current()` here:
https://github.com/dask/distributed/blob/0774365a305a22f05ff916318bdba57add3f2eb7/distributed/worker.py#L3498

[`IOLoop.current()`](https://github.com/tornadoweb/tornado/blob/790715ae0f0a30b9ee830bfee75bb7fa4c4ec2f6/tornado/ioloop.py#L264) calls [`asyncio.get_event_loop()`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop). Normally I believe this would be an error when not on the main thread, but because of our [`AnyThreadEventLoopPolicy`](https://github.com/dask/distributed/blob/0774365a305a22f05ff916318bdba57add3f2eb7/distributed/utils.py#L1069-L1078), we spin up a new, separate event loop for the calling thread.

We really should just be using [`asyncio.get_running_loop`](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_running_loop) for this case where we just want to check if there's already an event loop for the current thread, without creating one. But because we need a Tornado loop, not an asyncio loop, we can't just compare that to `loop` (though we could peek into [`IOLoop._ioloop_for_asyncio`](https://github.com/tornadoweb/tornado/blob/790715ae0f0a30b9ee830bfee75bb7fa4c4ec2f6/tornado/ioloop.py#L270)).

---

But more broadly, I'm wondering why we even have `AnyThreadEventLoopPolicy`. Worker threads spinning up their own event loops seems like a _very_ advanced use-case. Where do we actually need this behavior? Rather than adding this policy across the board, with potential for unexpected event loop creation like this, in those places where we do need it, could we explicitly create a new loop when necessary?

The `AnyThreadEventLoopPolicy` was introduced way back in https://github.com/dask/distributed/pull/1518. There was discussion of the issues with this on https://github.com/tornadoweb/tornado/issues/2183. Funnily enough, there was a [comment](https://github.com/tornadoweb/tornado/issues/2183#issuecomment-354564156) back then predicting this issue:

> This won't lead to waste unless you have a lot of threads that sometimes do something asynchronous but spend most of their time without an event loop, but that would be a very strange pattern.

This is exactly the pattern we've now created with `get_client`.

cc @jcrist

### PR DESCRIPTION
**Pull request by gjoseph92**
Just curious to see what fails so far.

With #5310 and #5311, closes #5309 (and ensures `uvloop` will work with everything)

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `black distributed` / `flake8 distributed` / `isort distributed`

### CODE REVIEW

**Comment by fjetter** [comment]
FYI this custom eventloop policy is one of the things why we currently have to pin pytest-asyncio

**Comment by mrocklin** [comment]
cc @minrk

File: distributed/semaphore.py [hunk 1]
```diff
@@ -412,7 +412,7 @@ def __init__(
 
     @property
     def asynchronous(self):
-        return self.loop is IOLoop.current()
+        return self.loop is IOLoop.current(instance=False)
```
Comment (minrk) [inline distributed/semaphore.py:415]: `instance` arg is [deprecated in tornado 5](https://www.tornadoweb.org/en/stable/ioloop.html#tornado.ioloop.IOLoop.current), and minimum supported tornado is 6.0.3.

**Review by minrk — APPROVED** [review]
👍 to removing the policy. You can generally ensure a loop would exist in background threads yourself, without enforcing it via a policy, via the same logic applied in the policy.

```python
try:
   loop = asyncio.get_event_loop()
except RuntimeError:
    loop = asyncio.new_event_loop()
    asyncio.set_event_loop(loop)
```

Then your code always behaves as desired, regardless of the configured policy.

**Comment by jcrist** [comment]
I've run into an issue that would be best fixed by this getting in.

We currently rely on `AnyThreadEventLoopPolicy` to automagically create a new event loop in a thread if it doesn't already exist. BUT. Since we use `pytest-asyncio`, which unconditionally resets the event loop policy at the end of any test decorated with `@pytest.mark.asyncio`, most of the time this event loop policy isn't even active in our test suite. This leads to all sorts of weird and hard to debug behavior where running a single test will work fine but running the whole suite will fail (since an earlier `pytest.mark.asyncio` test reset the event loop policy).

What would it take to get this PR in? The `instance` kwarg to `IOLoop.current` is marked as deprecated, but doesn't actually warn and has remained constant for the last 3 years in the sourcecode. Relying on it may not be ideal, but also doesn't seem that bad to me. We could also split that code out so we manually check `asyncio.get_event_loop()` trying `IOLoop.current()` if that'd be better.

**Comment by jcrist** [comment]
> FYI this custom eventloop policy is one of the things why we currently have to pin pytest-asyncio

Oop, just noticed that this is apparently a known issue. Spent too long debugging weird failures for no reason then :/.

**Comment by gjoseph92** [comment]
Done in https://github.com/dask/distributed/pull/5531

**Pull request closed without merging**